tools/extra/opae.io/opae/io/utils.py

```diff
@@ -281,12 +281,25 @@ def dfh_walk(region, offset=0, header=dfh0, guid=None):
             break
         offset += h.bits.next
 
-def walk(region, offset=0, show_uuid=False):
-    for offset_, hdr in dfh_walk(region, offset=offset):
+def walk(region,
+         offset=0, show_uuid=False, count=None, delay=None, dump=False):
+    for (offset_, hdr), i in enumerate(dfh_walk(region, offset=offset)):
+        if count and i >= count:
+            break
         print(f'offset: 0x{offset_:04x}, value: 0x{hdr.value:04x}')
         print(f'    dfh: {hdr}')
         if show_uuid:
             print(f'    uuid: {read_guid(region, offset_+0x8)}')
+            if delay:
+                time.sleep(delay)
+        if delay:
```

Comment:
Can the delay at 293 be omitted, because we're delaying here again?

Reply:
I had intended a read after every mmio read but I think I have to have a delay (for the header) before the delay for reading the uuid.